### PR TITLE
pprz_algebra_float.h new functions

### PR DESCRIPTION
**Pull request by coppolam**
Added a function to invert any matrix and a function to create an identity matrix.

### CODE REVIEW

**Comment by gautierhattenberger** [comment]
The new function are considering matrix as 1D arrays, while existing functions are using 2D arrays. I suggest to make them consistent.

**Comment by kirkscheper** [comment]
@gautierhattenberger are there any standards in pprz as to how matricies are defined. Most functions assume a matrix is defined as a double array (a[][]) e.g. float_mat_mul(float **o, float **a, float **b, int m, int n, int l), but some assume a single array (a[]). e.g float_mat_inv_2d(float inv_out[4], float mat_in[4]).

It would appear that a matrix defined by a single array is assumed to be a square matrix, is that correct?

Additionally, what is the policy with inline functions, preferred or not?

**Comment by gautierhattenberger** [comment]
There is no string guidelines at the moment. Double arrays are used for arbitrary size matrices. As you said, it seems that some fixed-size matrix functions are using single array. We could stick to this for now. Which means that in the case of this PR, double arrays are preferred.

Regarding inline or not, I would say that if the function is short enough (3 to 4 lines) it can be inline, and a regular function otherwise. But this is really arbitrary...

**Comment by coppolam** [comment]
I will change them, although for the future I think it might useful to have both versions (double and single). Passing single arrays means that the functions can also be directly used with array operations, so that  functions like float_mat_copy, float_mat_sum could also be used not just for matrix double arrays, but also for arrays.

File: sw/airborne/math/pprz_algebra_float.c [hunk 1]
```diff
@@ -765,3 +765,56 @@ bool float_mat_inv_4d(float invOut[16], float mat_in[16])
...
+
+  /* Cut out the identity, which has now moved to the left side */
+  for(i = 0 ; i < n ; i++ ) {
+    for(j = n; j < 2*n; j++ ) {
+      o[i][j-n] = a[i][j];
+
+    }
+  }
+
+}
```
Comment (kirkscheper) [inline sw/airborne/math/pprz_algebra_float.c]: file ending

**Pull request merged**